builtin/credential/approle/path_login.go

```diff
@@ -33,7 +33,11 @@ func pathLogin(b *backend) *framework.Path {
 // Returns the Auth object indicating the authentication and authorization information
 // if the credentials provided are validated by the backend.
 func (b *backend) pathLoginUpdate(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-	role, roleName, metadata, err := b.validateCredentials(req, data)
+	if req.Connection == nil || req.Connection.RemoteAddr == "" {
+		return nil, fmt.Errorf("failed to get connection information")
```

Comment:
Shouldn't this only be an error if the role/secret ID contains a bind against a CIDR? It seems like the right place to do this is in `validateCredentials`, and only after checking for a bind.